scripts/lib/releaseNotes.ts

```diff
@@ -150,3 +149,148 @@ function addNewReleaseNoteToc(releaseNotesNode: any, newVersion: string) {
+  // =>
+  // markdownByPatchSorted = [["0.46.0", "test 2"],["0.45.0", "test 1"]]
+  const markdownByPatchSorted = Object.entries(markdownByPatchVersion).sort(
+    ([version1], [version2]) => {
+      const versionPatch1 = version1.split("rc").slice(0, 1)[0];
+      const versionPatch2 = version2.split("rc").slice(0, 1)[0];
+
+      if (versionPatch1 == versionPatch2) {
+        // The release candidates within the same patch should appear last.
+        // e.g. version 0.45.1rc1 should appear after version 0.45.1.
+        if (version1.length < version2.length) {
+          return -1;
+        } else if (version1.length > version2.length) {
+          return 1;
+        }
+      }
+
+      return version2.localeCompare(version1);
+    },
+  );
+
+  return Object.fromEntries(markdownByPatchSorted);
+}
+
+/**
+ * Process the markdown dividing it into small sections for each patch version
+ *
+ * Returns a tuple:
+ *   1. A Set of strings representing all the minor versions we found in the
+ *      markdown.
+ *   2. A dictionary with the corresponding markdown for each patch version
+ *      found.
+ */
+export function extractMarkdownReleaseNotesPatches(
+  markdown: string,
+): [Set<string>, { [id: string]: string }] {
+  const sectionsSplit = markdown.split(/\n## (?=[0-9])/);
+  const sections: string[] = sectionsSplit.slice(1, sectionsSplit.length);
+
+  const minorVersionsFound = new Set<string>();
+  const markdownByPatchVersion: { [id: string]: string } = {};
+
+  for (let section of sections) {
+    const versionPatch = section.split("\n").slice(0, 1)[0];
+    const versionMinor = versionPatch.split(".").slice(0, 2).join(".");
+
+    minorVersionsFound.add(versionMinor);
+
+    const content = section.split("\n");
+    content.shift();
+    markdownByPatchVersion[versionPatch] = `## ${versionPatch}\n${content.join(
+      "\n",
+    )}`;
+  }
+
+  return [minorVersionsFound, markdownByPatchVersion];
+}
+
+/**
+ * Updates the release notes folder by adding the notes to their corresponding version
+ * file.
+ */
+export async function writeSeparateReleaseNotes(
+  pkg: Pkg,
+  releaseNoteMarkdown: string,
+): Promise<void> {
+  if (!pkg.hasSeparateReleaseNotes) {
+    throw new Error(
+      `The package ${pkg.name} doesn't have separate release notes`,
+    );
+  }
+
+  // Dictionary to store the file header in case we need to reconstruct a file from a
+  // previous version
+  const filesToInitialHeaders: { [id: string]: string } = {};
+  const basePath = `${getRoot()}/docs/api/${pkg.name}/release-notes`;
+
+  const [minorVersionsFound, markdownByPatchVersion] =
+    extractMarkdownReleaseNotesPatches(releaseNoteMarkdown);
+
+  // Read the current release notes for each version found
+  for (let version of minorVersionsFound) {
+    const versionPath = `${basePath}/${version}.md`;
+
+    if (!(await pathExists(versionPath))) {
+      // We don't have any release note file for that version
+      continue;
+    }
```

Comment:
Okay, I think I finally get it 😅

```suggestion
    // When we're adding a version, its release note file will not yet exist, so we
    // cannot read it here to determine the prior contents. Instead, skip the relese
    // note since we are going to add it later in updateApiDocs.ts.
    if (!(await pathExists(versionPath))) {
      continue;
    }
```